impl/src/test/kotlin/com/android/declarative/internal/agp/ApplicationVariantApiTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.android.declarative.internal.agp
+
+import com.android.build.api.dsl.ApplicationExtension
+import com.android.build.api.variant.AndroidComponentsExtension
+import com.android.build.api.variant.ApplicationAndroidComponentsExtension
+import com.android.build.api.variant.ApplicationVariant
+import com.android.build.api.variant.ApplicationVariantBuilder
+import com.android.build.api.variant.VariantSelector
+import com.android.declarative.internal.fixtures.FakeAndroidComponentsExtension
+import com.android.declarative.internal.variantApi.AndroidComponentsParser
+import org.gradle.api.provider.Property
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.tomlj.Toml
+
+class ApplicationVariantApiTest: AgpDslTest() {
+
+    @Mock
+    lateinit var variantBuilder: ApplicationVariantBuilder
+
+    @Mock
+    lateinit var variant: ApplicationVariant
+
+    @Mock
+    lateinit var selector: VariantSelector
+
+    private val extension: AndroidComponentsExtension<ApplicationExtension, ApplicationVariantBuilder, ApplicationVariant> by lazy {
+        FakeAndroidComponentsExtension(
+            selector,
+            variantBuilder,
+            variant
+        )
+    }
+
+    @Test
+    fun testVariantBuilderEnable() {
+
+        Mockito.`when`(selector.withName("debug")).thenReturn(selector)
+
+        val toml = Toml.parse(
+            """
+            [androidComponents."beforeVariants.debug"]
```

Comment:
why do we need quotes? without quotes seems better to me.

Reply:
the problem is that 
[androidComponents.beforeVariants]
enable = false

is the way to express something that applies to all variants which can also be expressed as 

androidComponents.beforeVariants..enable = false

therefore, when a find a dotted key for onVariants or beforeVariants, I cannot know if the key following onVariants is a property (like enable) or a selector (like debug).  I could play tricks like if there is only one subkey after `onVariants` and its one of the variant name then it probably is a selector but it's also fragile. 

Coming back to my example, if a variant is called 'enable' then it becomes hairy (I know the chances are really low).

So using quotes allow me to differentiate between the key part and a property name.
This is just the first version, I am planning to see if this can be improved in a subsequent CL.

Reply:
I can imaging we can do
[androidComponents.beforeVariants.all]
enable = false
for all variants.
And following for selecting by name
[androidComponents.beforeVariants."debug"]
enable = false

Reply:
that's a good idea but I wonder why "debug" is in quotes here ? we could just do 

[androidComponents.beforeVariants.debug]

Reply:
quoted debug means selector by name(regexp?)